### ISSUE TITLE
Add support for recording annotation type

### ISSUE THREAD

**Issue by dosumis** [cellannotation/cell-annotation-schema#29]
Some author categories record cell state rather than cell type.  Others record a mix of the two, or some other annotation relating to cell properties.  It would be useful to be able to systematically record this - maybe at the level of labelsets.

**Comment by dosumis** [comment]
Once this part of the standard is agreed, we should align the behaviour of pandasaurus_cxg with it.

**Comment by evanbiederstedt** [comment]
**Initial thoughts:**

- I recognize the value of this. Maybe simply asking users this question would be useful...
- This is such a massive fight, I really don't want to get into it....
- At some point, I think people will recognize there isn't a major difference between "a cell type" and a "cell state" in transcriptomic space.

**Comment by dosumis** [comment]
I would like to support this in the standard whether or not CAP chooses to adopt it.  It will remain optional.  There are clear cases of distinct state and type annotations in CxG and (e.g. whether cells are dividing or not) I would like to be able to distinguish them.

**Comment by dosumis** [comment]
> At some point, I think people will recognize there isn't a major difference between "a cell type" and a "cell state" in transcriptomic space.

I disagree (although to be clear we'd both need to define our terms carefully) - but that is a much bigger discussion.
